### PR TITLE
Add FD diagnostics to apply

### PR DESCRIPTION
**Pull request by mrzta**
## Context

Add diagnostics logging to apply app 

## Changes proposed in this pull request

Enable diagnostics logging and set default value 

## Guidance to review

Since our front doors are in the prod subscription you will require PIM
Clone the branch and run domains-infra-plan confirm changes 

## Link to Trello card

https://trello.com/c/8tCQhHrW/409-front-door-logging

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: terraform/custom_domains/infrastructure/main.tf [hunk 1]
```diff
@@ -3,4 +3,6 @@ module "domains_infrastructure" {
   hosted_zone            = var.hosted_zone
   tags                   = var.tags
   deploy_default_records = var.deploy_default_records
+  azure_enable_monitoring= var.azure_enable_monitoring
```
Comment (saliceti) [inline terraform/custom_domains/infrastructure/main.tf:6]: Can you format document?

File: terraform/custom_domains/infrastructure/variables.tf [hunk 2]
```diff
@@ -16,3 +16,8 @@ variable "delegation_ns" {
 variable "deploy_default_records" {
   default = true
 }
+
+variable "azure_enable_monitoring" {
+  type = bool
```
Comment (saliceti) [inline terraform/custom_domains/infrastructure/variables.tf:21]: Can you format document?

**Comment by mrzta** [comment]
[Front door logging](https://trello.com/c/8tCQhHrW/409-front-door-logging)

File: terraform/custom_domains/infrastructure/main.tf [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module "domains_infrastructure" {
-  source                 = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/infrastructure?ref=0.5.1"
-  hosted_zone            = var.hosted_zone
-  tags                   = var.tags
-  deploy_default_records = var.deploy_default_records
+  source                  = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/infrastructure?ref=0.5.1"
```
[the remark this answers is not in the document]
Comment (saliceti) [reply terraform/custom_domains/infrastructure/main.tf:2]: As discussed, this was released to terraform-modules testing tag, please change the ref to testing

[the remark this answers is not in the document]
Comment (mrzta) [reply terraform/custom_domains/infrastructure/main.tf]: ref changed to testing

**Review by saliceti — APPROVED** [review]

**Pull request merged**